keanu-project/src/main/java/io/improbable/keanu/network/NetworkSnapshot.java

```diff
@@ -32,10 +32,10 @@ private NetworkSnapshot(Collection<Vertex> vertices) {
      */
     public void apply() {
         for (Vertex v : values.keySet()) {
-            v.unobserve();
             if (observed.contains(v)) {
                 v.observe(values.get(v));
             } else {
+                v.unobserve();
```

Comment:
Nice change.  Just FMI - did you actually see a problem because of this, or is this just a code cleanup thing?

Reply:
Was causing problems with the `ModelVertex` and it ended up being this unneccessary observe!